Yoast/wordpress-seo
# Only mark subheading title if subheading text is too long

## Context


* 

## Summary


This PR can be summarized in the following changelog entry:

* Marks only the subheading when a subheading text is too long.
* [yoastseo] Adds an additional property to the `Mark` object, which can check which field to apply the marking to.
* [yoastseo] Only marks the subheading when a subheading text is too long in _subheading distribution_ assessment.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**NOTE**: classic and default editors need to be tested fully. In Gutenberg smoke-test suffices. Taxonomies and Elementor are excluded, as highlighting is not available there.  

#### Scenario 1: Test with one subheading
1. Install and activate YoastSEO (Free).
2. Set the site language to English
3. Create a post/page/CPT.
4. Add an H2 subheading.
5. Add a too long text below the subheading (more than 300 words).
6. Add the specific string from the subheading in the text below the subheading (note: it has to be absolutely the same, incl. capitalisation).
7. Toggle the eye marker for `Subheading distribution` assessment.
8. Make sure only the subheading is marked and NOT the 'subheading strings' in the text.
9. Change the H2 subheading to H3
10. Toggle the eye marker for `Subheading distribution` assessment.
8. Make sure only the subheading is marked and NOT the 'subheading strings' in the text.
11. Smoke test markers for other assessments that have highlights.

#### Scenario 2: Test with multiple subheadings
1. Install and activate YoastSEO (Free).
2. Set the site language to English
3. Create a post/page/CPT
4. Make the text length before the first subheading less than 300 words
5. Add an H2 subheading
6. Make the text length after the H2 subheading between 300-350 words. Also include the H2 Subheading string in the text (note: it has to be absolutely the same, incl. capitalisation).
7. Add an H3 subheading
8. Make the text length after the H3 subheading between 300-350 words
9. Toggle the eye marker for `Subheading distribution` assessment.
10.  Make sure that the H2 subheading and H3 subheading are marked and NOT the 'H2 subheading strings' in the text. 

#### Scenario 3: Block editor
* Repeat scenario 1 and 2

#### Scenario 4: Classic editor
* Repeat scenario 1 and 2
* Also test Woo product.

#### Scenario5: Shopify
* Test together with this PR: https://github.com/Yoast/shopify-seo/pull/885
   * The `shopify-seo` branch is `PC-887-if-subheading-text-too-long-only-mark-subheading-title`
   * Don't forget to link the `wordpress-seo` branch to the Shopify branch. See [this doc](https://github.com/Yoast/shopify-seo/blob/main/doc/run-the-app.md#embedded-development-of-admin-ui--wordpress-seo-packages) on how to do it.
* Repeat Scenario 1 and 2 and test fully for one content type in Shopify
* Smoke test in other content types

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* Make sure that all other assessments that use markings still work.

## UI changes


## Other environments

* [] This PR also affects Shopify. I have added a changelog entry starting with `[shopify-seo]`, added test instructions for Shopify and attached the `Shopify` label to this PR.

## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/PC-887
